src/commands/prepare_release/command.rs

```diff
@@ -321,18 +321,17 @@ fn promote_changelog_unreleased_to_version(
     changelog: &Changelog,
     version: &BuildpackVersion,
     date: &DateTime<Utc>,
-    updated_dependencies: &[BuildpackId],
+    updated_dependencies: &HashSet<BuildpackId>,
 ) -> Changelog {
     let updated_dependencies_text = if updated_dependencies.is_empty() {
         None
     } else {
-        Some(
-            updated_dependencies
-                .iter()
-                .map(|id| format!("- Updated `{id}` to `{version}`."))
-                .collect::<Vec<_>>()
-                .join("\n"),
-        )
+        let mut updated_dependencies_bullet_points = updated_dependencies
+            .iter()
+            .map(|id| format!("- Updated `{id}` to `{version}`."))
+            .collect::<Vec<_>>();
+        updated_dependencies_bullet_points.sort();
+        Some(updated_dependencies_bullet_points.join("\n"))
```

Comment:
FWIW, the code you posted wouldn't work because the return value of the `sort` would be what the expression would evaluate to - which is `()`. We'd need to add a `dependencies` at the end to make sure the expression evals to the correct value.

I think in this specific case it adds more noise to the code without any direct benefit, especially since the scope for `updated_dependencies_bullet_points` ends immediately afterwards.

I sometimes use the syntax you show here, but I often realize afterwards that the code could be a function, adding even more isolation to it and making the code easier to read since the logic has a name now.

Reply:
TIL i learned you could scope the variable like that @schneems.